src/lib/structures/symbols.ml

```diff
@@ -317,82 +296,242 @@ let name_to_string =
+    | Op Mult -> Fmt.pf ppf "*"
+    | Op Div -> Fmt.pf ppf "/"
+    | Op Modulo -> Fmt.pf ppf "%%"
+    | Op Pow -> Fmt.pf ppf "**"
+    | Op Sqrt_real -> Fmt.pf ppf "sqrt_real"
+    | Op Sqrt_real_default -> Fmt.pf ppf "sqrt_real_default"
+    | Op Sqrt_real_excess -> Fmt.pf ppf "sqrt_real_excess"
+    | Op Int_floor -> Fmt.pf ppf "int_floor"
+    | Op Int_ceil -> Fmt.pf ppf "int_ceil"
+    | Op Max_real -> Fmt.pf ppf "max_real"
+    | Op Max_int -> Fmt.pf ppf "max_int"
+    | Op Min_real -> Fmt.pf ppf "min_real"
+    | Op Min_int -> Fmt.pf ppf "min_int"
+    | Op Integer_log2 -> Fmt.pf ppf "integer_log2"
+    | Op Integer_round -> Fmt.pf ppf "integer_round"
+
+    (* Reals_Ints theory *)
+    | Op Abs_int -> Fmt.pf ppf "abs_int"
+    | Op Abs_real -> Fmt.pf ppf "abs_real"
+    | Op Real_of_int -> Fmt.pf ppf "real_of_int"
+    | Op Real_is_int -> Fmt.pf ppf "real_is_int"
+
+    (* FixedSizedBitVectors theory *)
+    | Bitv (n, s) ->
+      Fmt.pf ppf "[|%s|]" (Z.format (Fmt.str "%%0%db" n) s)
+    | Op Extract (i, j) ->
+      Fmt.pf ppf "^{%d; %d}" i j
+    | Op Concat -> Fmt.pf ppf "@"
+    | Op BV2Nat -> Fmt.pf ppf "bv2nat"
+    | Op Int2BV n -> Fmt.pf ppf "int2bv[%d]" n
+    | Op BVnot -> Fmt.pf ppf "bvnot"
+    | Op BVand -> Fmt.pf ppf "bvand"
+    | Op BVor -> Fmt.pf ppf "bvor"
+
+    (* ArraysEx theory *)
+    | Op Get -> Fmt.pf ppf "get"
+    | Op Set -> Fmt.pf ppf "set"
+
+    (* DT theory *)
+    | Op Record -> Fmt.pf ppf "@Record"
+    | Op (Access s) -> Fmt.pf ppf "@Access_%a" Hstring.print s
+    | Op (Constr s) -> Hstring.print ppf s
+    | Op (Destruct (s, g)) ->
+      Fmt.pf ppf "%s%a" (if g then "" else "!") Hstring.print s
+
+    (* Float theory *)
+    | Op Float -> Fmt.pf ppf "float"
+    | Op Fixed -> Fmt.pf ppf "fixed"
+
+    (* Symbols used in semantic triggers *)
+    | In (lb, rb) ->
+      Fmt.pf ppf "%s, %s" (string_of_bound lb) (string_of_bound rb)
+    | MapsTo x -> Fmt.pf ppf "%s |->" (Var.to_string x)
+    | Op Not_theory_constant -> Fmt.pf ppf "not_theory_constant"
+    | Op Is_theory_constant -> Fmt.pf ppf "is_theory_constant"
+    | Op Linear_dependency -> Fmt.pf ppf "linear_dependency"
+
+    | Lit lit -> pp_lit ppf lit
+    | Form form -> pp_form ppf form
+    | Let -> Fmt.pf ppf "let"
+
+    | Op (Optimize {order; is_max=true}) ->
+      Fmt.pf ppf "maximize(-,%d)" order
+    | Op (Optimize {order; is_max=false}) ->
+      Fmt.pf ppf "minimize(-,%d)" order
+
+    | Op Reach -> assert false
+end
+
+module SmtPrinter = struct
+  let pp_lit ppf lit =
+    match lit with
+    | L_eq -> Fmt.pf ppf "="
+    | L_neg_eq -> Fmt.pf ppf "distinct"
+    | L_built LE -> Fmt.pf ppf "<="
+    | L_built LT -> Fmt.pf ppf "<"
+    | L_neg_built LE -> Fmt.pf ppf ">"
+    | L_neg_built LT -> Fmt.pf ppf ">="
+    | L_neg_pred -> Fmt.pf ppf "not"
+    | L_built (IsConstr hs) -> Fmt.pf ppf "(_ is %a)" Hstring.print hs
+    | L_neg_built (IsConstr hs) -> Fmt.pf ppf "(not (_ is %a))" Hstring.print hs
```

Comment:
This is not a valid smt-lib symbol?

Reply:
Which one?

Reply:
The one on the line that I commented on :) `(not (_ is %a))`